tokenizers/src/models/mod.rs

```diff
@@ -33,7 +33,17 @@ impl<'a> Serialize for OrderedVocabIter<'a> {
     where
         S: Serializer,
     {
-        let iter = (0u32..(self.vocab_r.len() as u32)).map(|i| (&self.vocab_r[&i], i));
+        // There could be holes so max + 1 is more correct than vocab_r.len()
+        let max = self.vocab_r.iter().map(|(key, _)| key).max().unwrap_or(&0) + 1;
+        let iter = (0..max).filter_map(|i| {
+            if let Some(token) = self.vocab_r.get(&i){
+                Some((token, i))
+            }else{
+                warn!("The OrderedVocab you are attempting to save contains a hole for index {}, your vocabulary could be corrupted !", i);
+                println!("The OrderedVocab you are attempting to save contains a hole for index {}, your vocabulary could be corrupted !", i);
```

Comment:
why are `warn!` & `println!` both necessary? Because of bindings?

Reply:
`warn!` won't print anything if the logger is not set (the default in rust).

Since the extreme level of concern this behavior should cause I think I want to really yell even if the `warn!` `debug!` thing is not setup.

Highly debattable.